app/tasks/maintenance_tasks/task.rb

```diff
@@ -103,13 +141,36 @@ def load_constants
       end
     end
 
-    # Placeholder method to raise in case a subclass fails to implement the
-    # expected instance method.
+    # The contents of a CSV file to be processed by a Task.
     #
-    # @raise [NotImplementedError] with a message advising subclasses to
-    #   implement an override for this method.
+    # @return [String] the content of the CSV file to process.
+    def csv_content
+      raise NoMethodError unless has_csv_content?
+
+      @csv_content
+    end
+
+    # Set the contents of a CSV file to be processed by a Task.
+    #
+    # @param csv_content [String] the content of the CSV file to process.
+    def csv_content=(csv_content)
+      raise NoMethodError unless has_csv_content?
+
+      @csv_content = csv_content
+    end
+
+    # Returns whether the Task handles CSV.
+    #
+    # @return [Boolean] whether the Task handles CSV.
+    def has_csv_content?
+      self.class.has_csv_content?
```

Comment:
You can `delegate :has_csv_content?, to: :class` here.

Reply:
Yes, but I particularly think `delegate` is overused and prefer the clear method. It is easier to refactoring and to search in the codebase.